nifi-registry/nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/service/RegistryService.java

```diff
@@ -339,10 +339,12 @@ public VersionedFlow createFlow(final String bucketIdentifier, final VersionedFl
         if (versionedFlow.getBucketIdentifier() == null) {
             versionedFlow.setBucketIdentifier(bucketIdentifier);
         }
+        if (!preserveSourceProperties || (versionedFlow.getCreatedTimestamp() <= 0) || (versionedFlow.getModifiedTimestamp() <= 0)) {
```

Comment:
Actually this approach could be changed so that if the caller sets the timestamps, then they will be preserved, instead of passing the `preserveSourceProperties` argument to this service level. The checks should also be limited to the created timestamp so that the modified timestamp is always updated.
```suggestion
        if (versionedFlow.getCreatedTimestamp()) {
```